### ISSUE TITLE
DROID dataset misalignment (Gripper dim 8)

### ISSUE THREAD

**Issue by kianhk6** [Dexmal/realtime-vla#27]
When I run the test.py for the droid tuned checkpoint I get a decent amount of discrepancy 
The gripper seems to be having issues.

Results: (seed 400)

<img width="2211" height="1745" alt="Image" src="https://github.com/user-attachments/assets/64b5140a-e400-4bed-81f4-f6c806ca650c" />

GPU: L40s (still same architecture as 4090 but higher VRAM)

**Comment by kianhk6** [comment]
My pi05_infer.py (I believe the masking is missing since for droid they mask one of the views but not sure how to add it) 
Plus fixed the action dim and added seeding for the test file. I made sure also the preprocessing is same post processing.

[test.py](https://github.com/user-attachments/files/24987082/test.py)

[pi05_infer.py](https://github.com/user-attachments/files/24987086/pi05_infer.py)

**Comment by kianhk6** [comment]
Looking at the norm_stats:
Joints 1-7: Symmetric ranges like [-0.46, 0.45] centered around 0
Gripper (8th): q01=0.0, q99=1.0 - asymmetric, all positive

So this can be the issue

The checkpoint is from openpi itself gs://openpi-assets/checkpoints/pi05_droid

**Comment by Tomorin523** [comment]
> My pi05_infer.py (I believe the masking is missing since for droid they mask one of the views but not sure how to add it) Plus fixed the action dim and added seeding for the test file. I made sure also the preprocessing is same post processing.
> 
> [test.py](https://github.com/user-attachments/files/24987082/test.py)
> 
> [pi05_infer.py](https://github.com/user-attachments/files/24987086/pi05_infer.py)

[test.py](https://github.com/user-attachments/files/25000912/test.py) we don't use mask, instead we discard the unused input, please try this file. I just change the num_views 
```python
        if self.model_version == "pi05":
            policy = Pi05Inference(
                checkpoint = weights,
                num_views = 2,
                chunk_size = self.action_horizon,
                tokenizer_path=self.tokenizer_path,
                max_tokenize_len=200,
                max_prompt_text = self.prompt,
                discrete_state_input = self.discrete_state_input,
                state_dim_for_max_prompt = self.state_len
            )

```
and the image inputs
```python
    def infer(self, inputs: dict, noise: np.ndarray) -> dict:
        if self.model_type == "triton":
                                                    .........
            transformed_inputs = self._apply_input_transforms(inputs, norm_stats = self.norm_stats)
            imgs = []
            for view in ["base_0_rgb", "left_wrist_0_rgb"]:
                img = transformed_inputs["image"][view]
                imgs.append(torch.from_numpy(img))
            observation_images = torch.stack(imgs, dim=0).to(torch.float32).cuda(non_blocking=True)

```

**Comment by kianhk6** [comment]
<img width="2211" height="1745" alt="Image" src="https://github.com/user-attachments/assets/6dbb7368-c952-45d4-965b-52808104c6c9" />

Much better thank you!

**Comment by Tomorin523** [comment]
Thank you for the issue, we'll give more detailed instructions for different num_views in the next version.
